proto/vtgate.proto

```diff
@@ -252,6 +252,46 @@ message ExecuteEntityIdsResponse {
   query.QueryResult result = 3;
 }
 
+// ExecuteBatchRequest is the payload to ExecuteBatch.
+message ExecuteBatchRequest {
+  // caller_id identifies the caller. This is the effective caller ID,
+  // set by the application to further identify the caller.
+  vtrpc.CallerID caller_id = 1;
+
+  // session carries the current transaction data. It is returned by Begin.
+  // Do not fill it in if outside of a transaction.
+  Session session = 2;
+
+  // query is the query and bind variables to execute.
+  repeated query.BoundQuery query = 3;
+
+  // tablet_type is the type of tablets that this query is targeted to.
+  topodata.TabletType tablet_type = 4;
+
+  // not_in_transaction is deprecated and should not be used.
+  bool not_in_transaction = 5;
+
+  // keyspace to target the query to.
+  string keyspace = 6;
+
+  // options
+  query.ExecuteOptions options = 7;
+}
```

Comment:
How about adding as_transaction in there too? Even if we don't support it in all cases just yet...